core/src/main/scala/org/apache/gearpump/cluster/client/ClientContext.scala

```diff
@@ -53,20 +55,21 @@ class ClientContext(config: Config, sys: ActorSystem, _master: ActorRef) {
   }
 
   private val LOG: Logger = LogUtil.getLogger(getClass)
-  private implicit val timeout = Timeout(5, TimeUnit.SECONDS)
-
   implicit val system = Option(sys).getOrElse(ActorSystem(s"client${Util.randInt()}", config))
   LOG.info(s"Starting system ${system.name}")
-  val shouldCleanupSystem = Option(sys).isEmpty
-
+  private val shouldCleanupSystem = Option(sys).isEmpty
```

Comment:
The `system` won't be empty but `sys` can still be empty. `sys` not being empty means user has provided a ActorSystem so clientcontext should clean up user's actor system when being closed.